jax/_src/interpreters/partial_eval.py

```diff
@@ -1808,15 +1809,18 @@ def _const_folding_and_forwarding(
   consts: dict[Var, Any] = dict(zip(jaxpr.constvars, constvals))
   var_subs: dict[Var, Var] = {}  # not Dict[Var, Atom] b/c literals not inlined
   new_eqns = []
+  def apply_var_sub(a: Atom) -> Atom:
+    return var_subs.get(v, v) if isinstance(v, Var) else v
```

Comment:
`v` is undefined

Reply:
or worse... is defined in the global scope 😬

Reply:
Good catch, thanks